contracts/sky/src/handle.rs

```diff
@@ -0,0 +1,270 @@
+use cosmwasm_std::{
+    Storage, Api, Querier, Extern, Env, StdResult, HandleResponse, to_binary, 
+    StdError, HumanAddr, CosmosMsg, Binary, WasmMsg
+};
+use cosmwasm_math_compat::Uint128;
+use shade_protocol::{
+    utils::{asset::Contract},
+    contract_interfaces::{
+    sky::sky::{
+        Config, HandleAnswer, self
+    },
+    dex::sienna::{PairQuery, TokenTypeAmount, PairInfoResponse, TokenType, Swap, SwapOffer, CallbackMsg, CallbackSwap},
+    mint::mint::{QueryAnswer, QueryMsg, QueryAnswer::Mint, HandleMsg::Receive, self},  
+    snip20::Snip20Asset,
+}};
+use secret_toolkit::utils::Query;
+use crate::{state::config_r, query::trade_profitability};
+
+pub fn try_update_config<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    config: Config,
+) -> StdResult<HandleResponse> {
+    Ok(HandleResponse{
+        messages: vec![],
+        log: vec![],
+        data: Some(to_binary(&HandleAnswer::UpdateConfig{
+            status: true,
+        })?),
+    })
+}
+
+pub fn try_arbitrage_event<S: Storage, A: Api, Q: Querier>(
+    deps: &mut Extern<S, A, Q>,
+    env: Env,
+    amount: Uint128,
+) -> StdResult<HandleResponse> {
+    let config: Config = config_r(&deps.storage).load()?;
+    let pool_info: PairInfoResponse = PairQuery::PairInfo.query(
+        &deps.querier,
+        env.contract_code_hash.clone(),
+        config.market_swap_addr.address.clone(),
+    )?;
+    let test_amount: u128 = 100000000;
+    let mint_info: QueryAnswer = QueryMsg::Mint{
+        offer_asset: config.shd_token.contract.address.clone(),
+        amount: Uint128::new(test_amount),
+    }.query(
+        &deps.querier,
+        env.contract_code_hash.clone(),
+        config.mint_addr.address.clone(),
+    )?;
+    let mut mint_price: Uint128 = Uint128::new(0);
+    match mint_info{
+        QueryAnswer::Mint {
+            asset,
+            amount,
+        } => {
+            mint_price = amount;
+        },
+        _ => {
+            mint_price = Uint128::new(0);
+        },
+    };
+    let mut nom = Uint128::new(0);
+    let mut denom = Uint128::new(0);
+    if pool_info.pair_info.amount_0.u128().lt(&pool_info.pair_info.amount_1.u128()) {
+        nom = Uint128::new(pool_info.pair_info.amount_1.u128().clone() * 100000000);
```

Comment:
Always try to avoid possible overflows
```suggestion
        nom = pool_info.pair_info.amount_1.checked_mul(Uint128::new(100000000))?;
```